src/scratchOrgInfoApi.ts

```diff
@@ -353,33 +355,50 @@ export const pollForScratchOrgInfo = async (
   const logger = await Logger.child('scratchOrgInfoApi-pollForScratchOrgInfo');
   logger.debug(`PollingTimeout in minutes: ${timeout.minutes}`);
 
-  const response = await retry(
-    async () => {
-      const resultInProgress = await hubOrg
-        .getConnection()
-        .sobject<ScratchOrgInfo>('ScratchOrgInfo')
-        .retrieve(scratchOrgInfoId);
-      logger.debug(`polling client result: ${JSON.stringify(resultInProgress, null, 4)}`);
-      // Once it's "done" we can return it
-      if (resultInProgress.Status === 'Active' || resultInProgress.Status === 'Error') {
-        return resultInProgress;
+  let resultInProgress!: ScratchOrgInfo;
```

Comment:
I avoid `!` where practical.  It's saying, "I 100% guarantee this will be defined" and prevents TS from reminding you to check for that to be the case.